packages/story-editor/src/components/form/linkIcon.js

```diff
@@ -53,6 +57,8 @@ function LinkIcon({ handleChange, icon, isLoading = false, ...rest }) {
     return message;
   }, [allowedImageFileTypes]);
 
+  const options = hasUploadMediaAction ? ['edit', 'remove'] : ['remove'];
```

Comment:
This capability check might be better handled directly in `StyledMedia` itself. Otheriwse every caller component has to check this, which is easily forgotten (see c92060657182730c19063ab9a0d0ba3910a38e47 for reference)

Reply:
Where else is the is a problem?

Reply:
Not necessarily saying it _is_ a problem, but it _could_ be, for example if someone adds a new feature using `MediaInput` and does not know that they need to add such a capability check.
It's also a bit repetitive because every consumer needs to add this check, whereas it could be done _just once_ within `MediaInput` itself.

Reply:
The reason it done here, is that it is valid to allow for removal of link icons. This is one example where we want to allow this behaviour, this is an edge case. In most cases we want to hide the icon. So the existing behaviour is correct.